File: src/leetcode/python/bit_manipulation/BitwiseANDOfNumbersRange_201.py
class BitwiseANDOfNumbersRange_201(object):
    def rangeBitwiseAnd(self, m, n):
        if m == 0: return 0
        shift = 0
        while (m != n):
            m >>= 1
            n >>= 1
            shift += 1
        return n << shift

File: src/leetcode/java/medium/BitwiseANDOfNumbersRange_201.java
package leetcode.java.medium;

public class BitwiseAndOfNumbersRange_201 {
  // The approach is to find the leftmost common digits of m and n
  public int rangeBitwiseAnd(int m, int n) {
    if (m == 0) return 0;
    int shift = 0;
    while (m != n) {
      m >>= 1;
      n >>= 1;
      shift++;
    }
    return n << shift;
  }
}
